### PR TITLE
ROX-18113: Create compliance by cluster widget

### PR DESCRIPTION
**Pull request by bradr5**
## Description

Create the compliance by cluster widget. Should display the percent of compliance by cluster.

Note: currently uses static data

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Screenshot 2023-07-31 at 3 05 46 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+} from 'utils/chartUtils';
+
+import {
+    LOW_SEVERITY_COLOR,
+    MODERATE_MEDIUM_SEVERITY_COLOR,
+    IMPORTANT_HIGH_SEVERITY_COLOR,
+    CRITICAL_SEVERITY_COLOR,
+} from 'constants/visuals/colors';
+
+import WidgetCard from '../../../Dashboard/Widgets/WidgetCard';
```
Comment (bradr5) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx:19]: Any objections to pulling this out of the Dashboard container?

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    defaultChartBarWidth,
+    patternflySeverityTheme,
+} from 'utils/chartUtils';
+
+import {
+    LOW_SEVERITY_COLOR,
+    MODERATE_MEDIUM_SEVERITY_COLOR,
+    IMPORTANT_HIGH_SEVERITY_COLOR,
+    CRITICAL_SEVERITY_COLOR,
+} from 'constants/visuals/colors';
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx:17]: Super to use the single source of truth for severity colors!

For your info, the file will move to more intuitive path constants/severityColors when we merge #7190

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        link: '',
+    },
+];
+
+function ComplianceByCluster() {
+    const [complianceData] = useState(mockComplianceData);
+    const [widgetContainer, setWidgetContainer] = useState<HTMLDivElement | null>(null);
+    const widgetContainerResizeEntry = useResizeObserver(widgetContainer);
+
+    function getBarColor(percent) {
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx]: Style question for your consideration about pro and con to move pure function from component function scope to file scope?

Also, classic compliance has a different mapping. It is a worthwhile question for PM and UX about pro and con to align 1.0 with 2.0 mapping similar to how we have aligned colors with Workflow CVEs.
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Compliance/widgets/colorsForCompliance.ts#L9-L23

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        }
+        return CRITICAL_SEVERITY_COLOR;
+    }
+
+    return (
+        <WidgetCard isLoading={false} header="Compliance by cluster">
+            <div ref={setWidgetContainer}>
+                <Chart
+                    ariaDesc="Compliance coverage percentages by cluster"
+                    ariaTitle="Compliance coverage by cluster"
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx:86]: Can you search online if whether there is any guideline for or against accessibility attributes when there is heading text?

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx (hunk 3 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx]: Good callout, I didn't even know that existed there. I'd hate to copy that file over and have a duplicate but I don't want to be importing anything from 1.0. Any suggestions?

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx:18]: Dave says to go for it. Can you do that as a follow up pull request?
Move WidgetCardEmptyState also at the same time

**Review by pedrottimark — APPROVED** [review]

**Pull request merged**